tests/link/c/test_cmodule.py

```diff
@@ -4,33 +4,90 @@
 But this one tests a current behavior that isn't good: the c_code isn't
 deterministic based on the input type and the op.
 """
+import contextlib
 import logging
 import multiprocessing
 import os
+import sys
 import tempfile
+import threading
 from unittest.mock import patch
 
 import numpy as np
 import pytest
 
 import aesara
+import aesara.compile.compiledir
+import aesara.link.c.cmodule
 import aesara.tensor as at
 from aesara.compile.function import function
 from aesara.compile.ops import DeepCopyOp
 from aesara.configdefaults import config
-from aesara.link.c.cmodule import GCC_compiler, default_blas_ldflags
+from aesara.link.c.cmodule import GCC_Compiler, default_blas_ldflags
 from aesara.link.c.exceptions import CompileError
 from aesara.tensor.type import dvectors
 
 
-class MyOp(DeepCopyOp):
-    nb_called = 0
+@contextlib.contextmanager
+def force_change_compile_dir(new_compile_dir):
+    # while testing, somebody might have compiled
+    # a similar graph and loaded the module. E.g
+    # test_inter_process_cache compiles a + b
+    # and
+    # test_import_precompiled_module compiles the same
+    # If not taken in account,
+    # the imports will conflict in aesara.link.c.cmodule.dlimport
+    # raising an uninformative message.
+    # So yeah, what is done in this test tries to avoid
+    # any sort of conflict with other tests
+    #
+    # Modules names are bound to keys.
+    # We can just add some random salt to module keys and avoid duplication
+    compiledir_prop = aesara.config._config_var_dict["compiledir"]
+    with patch.object(
+        compiledir_prop, "val", new_compile_dir, create=True
+    ), patch.object(
+        aesara.config, "get_config_hash", new=lambda: str(hash(new_compile_dir))
+    ):
+        os.makedirs(new_compile_dir, exist_ok=True)
+        yield str(new_compile_dir)
```

Comment:
This fixture is a great start to the necessary improvements mentioned earlier, but it doesn't account for all the objects affected by the initial, immutable compilation directory setting.  For instance, notice that we had to patch `aesara.link.c.cmodule._module_cache` as well.  That should probably be included here, too.

There may be a few other things that still need to be patched, but this fixture should at least start with everything of which we're currently aware.

Reply:
This property is removed in this PT I do not rely on python cache since everything is on disk deterministically